src/js/line.js

```diff
@@ -0,0 +1,521 @@
+        var dataSet = that.model.wrappedDataSet;
+
+        var minDates = [];
+        var maxDates = [];
+
+        fluid.each(dataSet, function (setItem) {
+            var minDate = d3.min(setItem.data, function (d) {
+                return new Date(d.date);
+            });
+            minDates.push(minDate);
+
+            var maxDate = d3.max(setItem.data, function (d) {
+                return new Date(d.date);
+            });
+            maxDates.push(maxDate);
+
+        });
+
+        var minDate = d3.min(minDates, function (d) {
+            return new Date(d);
+        });
+
+        var maxDate = d3.max(maxDates, function (d) {
+            return new Date(d);
+        });
+
+        return d3.time.scale()
+            .domain([minDate, maxDate])
+            .range([padding, width - padding * 2]);
+    };
+
+    floe.chartAuthoring.lineChart.chart.getYAxis = function (that) {
+        var width = that.options.svgOptions.width;
+        var padding = that.options.lineOptions.padding;
+        var yScale = floe.chartAuthoring.lineChart.chart.getYScale(that);
+
+        var yAxis = d3.svg.axis()
+            .orient("left")
+            .scale(yScale)
+            .innerTickSize(- width + padding * 3)
+            .outerTickSize(0)
+            .tickPadding(10);
+
+        return yAxis;
+    };
+
+    floe.chartAuthoring.lineChart.chart.getXAxis = function (that) {
+        var xScale = floe.chartAuthoring.lineChart.chart.getXScale(that),
+            numberOfXAxisTicks = that.options.lineOptions.numberOfXAxisTicks;
+
+        // See https://github.com/mbostock/d3/wiki/Time-Formatting for
+        // explanation of how time formatting works in D3
+        var customTickFormat = d3.time.format.multi([
+            [".%L", function (d) { return d.getMilliseconds(); }],
+            [":%S", function (d) { return d.getSeconds(); }],
+            ["%I:%M", function (d) { return d.getMinutes(); }],
+            ["%I %p", function (d) { return d.getHours(); }],
+            ["%a %d", function (d) { return d.getDay() && d.getDate() !== 1; }],
+            ["%b %d", function (d) { return d.getDate() !== 1; }],
+            ["%b", function (d) { return d.getMonth(); }],
+            ["%Y", function () { return true; }]
+        ]);
+
+        var xAxis = d3.svg.axis()
+            .tickFormat(customTickFormat)
+            .ticks(numberOfXAxisTicks)
+            .orient("bottom")
+            .scale(xScale);
+
+        return xAxis;
+
+    };
+
+    floe.chartAuthoring.lineChart.chart.getLineGenerator = function (that) {
+        var interpolation = that.options.lineOptions.interpolation,
+            yScale = floe.chartAuthoring.lineChart.chart.getYScale(that),
+            xScale = floe.chartAuthoring.lineChart.chart.getXScale(that);
+
+        var line = d3.svg.line()
+            .interpolate(interpolation)
+            .x(function (d) {
+                return xScale(new Date(d.date));
```

Comment:
All of this infrastructure would work better if the assumption weren't hard-coded into the impl that the x-axis represents time. Therefore all of the core APIs should just accept plain Numbers as arguments rather than wrapping them in Dates